blog/2020-09-angular-schematics-common-helpers.md

```diff
@@ -682,11 +685,179 @@ export function playground(_options: any): Rule {
+
+### Run Schematics by using the `schematic()` and `externalSchematic()` function
+
+Perfect, we can now execute and combine our Schematics.
+But what if we want to actually combine external Schematics developed by others and integrate them in our Schematic?
+As we are actually lazy we don't want to write in the docs that others have to execute some other things before and then run our Schematic.
+
+Imagine you are working in a big company with lot's of Angular projects that are ready to start.
+This company has its own UI library already and all the applications are very different and ran by different teams (so not really a use case for a Monorepo).
+But there are also things they all have in common like a Single-Sing-On.
+Also, the basic design looks always similar (at least the header and the footer of the apps).
+
+Often I've seen companies building a reference implementation for such apps that's cloned / copied and adjusted by all developers.
+But there are some problems with this kind of workflow:
+
+- You have to keep the reference project always up-to-date
+- You have to clean up your copy of the project from stuff you don't need
+- You need to tell all teams copying this reference to keep track of changes and to adjust their copies frequently
+
+So a better solution in my opinion is to use the Schematics.
+You can create a `ng new` schematic that will scaffold the whole project code for you.
+But you don't want to start from scratch.
+So you probably want to combine things like this:
+
+- `ng add`: Add your Schematic to existing projects
+  - Cooperate UI library (always)
+  - Single-Sign-On (optional)
+  - Header-Component (optional)
+  - Footer-Component (optional)
+- `ng new`: Create a new project with your company defaults
+  - Create the basic application generated from the Angular CLI (**`externalSchematic`**)
+  - Run the `ng add` schematic
+
+Alright, we already know how we can achieve the most of the things.
+But one thing we haven't learned yet: How to run other (external) Schematics?
+Therefore, we can use the `externalSchematic` function.
+
+But first things first, let's check if our collection file is ready to start:
+
+```json
+{
+  "$schema": "../node_modules/@angular-devkit/schematics/collection-schema.json",
+  "schematics": {
+    "ng-add": {
+      "description": "Call other Schematics from the same or other packages.",
+      "factory": "./ng-add/index#playground"
+    },
+    "ng-new": {
+      "description": "Execute `ng new` with predefined options and run other stuff",
+      "factory": "./ng-new/index#playground"
+    }
+  }
+}
+```
+
+> Using the special Schematic namen `ng-add` and `ng new` let's you later use the schematic with executing `ng add`/`ng new` (instead of other Schematics called with `ng generate`).
+> There is also the special Schematic name `ng-update`, that will be called in the end with the `ng update` Angular CLI command.
```

Comment:
```suggestion
> Using the special Schematic names `ng-add` and `ng-new` let's you later use the schematic by just executing `ng add`/`ng new` (instead of other Schematics called with `ng generate`).
> There is also a special Schematic named `ng-update` which will be called in the end with the `ng update` Angular CLI command.
```